src/Compilers/CSharp/Portable/Binder/Binder_Lookup.cs

```diff
@@ -1404,9 +1406,9 @@ private static bool IsDerivedType(NamedTypeSymbol baseType, NamedTypeSymbol deri
                 visited?.Free();
             }
 
-            if (baseType.IsInterface)
+            if (baseType is NamedTypeSymbol { IsInterface: true } && derivedType is NamedTypeSymbol derivedNamedType)
```

Comment:
>derivedType is NamedTypeSymbol derivedNamedType [](http://example.com/codeflow?start=69&length=47)

Consider checking ```IsInterface``` for this one too

Reply:
`derivedType` may not be an interface. The point of this change is just to adapt to the updated signature of `derivedType` (now a `TypeSymbol` instead of a `NamedTypeSymbol`)

Reply:
> Why is it useful to get into the if in such case?

Yes, consider:
```
interface I { }
class C : I { }
```
We call `IsDerivedType(baseType: "I", derivedType: "C")`. We want to look at the base interfaces of "C" to figure out whether "I" is part of that list.
In this case, we want `IsDerivedType` to return `true` ("C" is derived from "I").

If I add an `IsInterface` check to `derivedType`, a test like `PreferMoreSpecific_Static_InterfaceFieldAndMoreSpecificField` starts reporting an ambiguity between two extensions fields, even though on more specific.

Reply:
I do not think we should be doing this. Interfaces are not base classes for class types.

Reply:
At least this feels like something new and should be explicitly brought to LDM.

Reply:
After some thought, I think I agree that we want to have the behavior that you describe, however this behavior is somewhat surprising for a method named `IsDerivedType`. Consider renaming the method to "DerivesOrImplements" and renaming "baseType" to "baseTypeOrImplementedInterface"